static/js/components/NewInstrument.jsx

```diff
@@ -0,0 +1,106 @@
+import React from "react";
+import { useDispatch, useSelector } from "react-redux";
+import Form from "@rjsf/material-ui";
+import { showNotification } from "baselayer/components/Notifications";
+import { submitInstrument } from "../ducks/instrument";
+import { fetchInstruments } from "../ducks/instruments";
+
+const NewInstrument = () => {
```

Comment:
I managed to crash the page when posting an instrument. Something to do with form verification. I will try to recreate but I suggest you play around with it too, you may be able to shake something loose.

Reply:
Ok I think I got these. See what you think. There is now a REFRESH_INSTRUMENTS  call (I think), please let me know if it is what you had in mind.

Reply:
I think it looks ok now. Thanks!